pkg/util/k8sutil/tpr_util.go

```diff
@@ -110,8 +99,11 @@ func updateClusterTPRObject(k8s *unversioned.Client, host, ns string, e *spec.Et
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
+	return sendReq(k8s.Client, req)
+}
 
-	resp, err := k8s.Client.Do(req)
+func sendReq(httpcli *http.Client, req *http.Request) (*spec.EtcdCluster, error) {
```

Comment:
well... it is a strange method. it accepts a http req but generates a operator specific return (type etcdCluster).